src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MetadataHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // This static helper class adds common entities to a Metadata Builder.
+    internal static class MetadataHelper
+    {
+        internal static AssemblyReferenceHandle AddAssemblyReference(Assembly assembly, MetadataBuilder metadata)
+        {
+            AssemblyName assemblyName = assembly.GetName();
+
+            return AddAssemblyReference(metadata, assemblyName.Name!, assemblyName.Version, assemblyName.CultureName, assemblyName.GetPublicKey(), (AssemblyFlags)assemblyName.Flags);
```

Comment:
The assembly reference should only contain a subset of flags from the assembly definition. Most of the flags are implementation-specific and should not be copied over.

It should be something like this: https://github.com/dotnet/roslyn/blob/c94b8ca5bf8d4bb4a26150a883757600218f6c5e/src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs#L1951